### DISCUSSION TITLE
Get main window's position coordinates

### DISCUSSION THREAD

**Discussion by mrkaragoz** [emilk/egui#1708] [Q&A] [answered]
Hello guys. Is there a way to get the main window's current position coordinates? I would like to save them in a config file and start the app in the same position with ```initial_window_pos``` (without the persistence feature)

**Comment by mrkaragoz** [comment]
It could be something like this:
```rust
fn on_exit_event(&mut self) -> bool {
    MyApp::update_config_for_window_pos(something::something::something::get_current_window_pos());  // function to update config
    return true
}
```

**Comment by mrkaragoz** [comment, marked as answer]
In commit 3c685d7bf6a36bda9ffbee36ee822b4205b29b36 @TicClick added `WindowInfo`. This provides position and size information of the native window. This info available with `eframe::Frame::info().window_info`. As of writing this answer, it is not included in the latest version of `egui (0.18.1)` and `eframe (0.18.0)`. In order to use this feature update the `Cargo.toml` as follows:

```toml
[dependencies]
egui = { git = "https://github.com/emilk/egui"}
eframe = { git = "https://github.com/emilk/egui"}
```

After that, it is possible to:
```rust
impl eframe::App for MyApp {
    fn update(&mut self, ctx: &egui::Context, frame: &mut eframe::Frame) {
        let window_info = frame.info.window_info.clone();
    }
}
```

**Reply by TeamDman** [reply]
Looks like window_info is no longer present, but https://github.com/emilk/egui/issues/2696 suggests 
> use Context::screen_rect or InputState::screen_rect (same thing)
but that doesn't seem to have the position I was expecting, perhaps it's saying 0,0 because egui is drawing inside the whole window but that screen rect isn't aware of the OS window stuff, only it's area inside it
